main.go

```diff
@@ -122,13 +124,14 @@ func main() {
 		logger := cfg.createLogger()
 
 		ctx := context.Background()
-		awsQueryConfigs, err := cfg.buildAWSConfig(ctx, cfg.maxReadRetries)
+
+		awsQueryConfigs, err := cfg.buildAWSConfig(ctx, cfg.maxReadRetries, cfg.readBaseEndpoint)
```

Comment:
Can we follow the same naming conventions?
```suggestion
		awsQueryConfigs, err := cfg.buildAWSConfig(ctx, cfg.maxReadRetries, cfg.queryBaseEndpoint)
```

Reply:
errhm yeah so this bothered me a bit as well. I noticed that `Query` and `ingest` aligns more with AWS docs, but `read` and `write` is the convention in the repo. I agree that we want to be consistent and follow one convention, but replacing read/write with query/ingest across the codebase will be a bigger change. Should we stick with read/write or open an issue for doing a refactor at some point?

Reply:
We can choose to refactor everything later. For now, for this feature, let's align with the names of the clients, which are `WriteClient` and `QueryClient`.